clash-prelude/src/Clash/Explicit/Signal.hs

```diff
@@ -859,4 +861,88 @@ sampleWithResetN
 sampleWithResetN nReset nSamples f =
   take nSamples (sampleWithReset nReset f)
 
+-- | Simulate a component until it matches a condition
+--
+-- It prints a message of the form
+--
+-- "Signal sampled for N cycles until value X"
+--
+-- A common usage is with a test bench using
+-- 'Clash.Explicit.Testbench.outputVerifier'. __NB__: Since this uses
+-- 'Clash.Explicit.Testbench.assert', when using @clashi@, read the
+-- note at "Clash.Explicit.Testbench#assert-clashi".
+--
+-- __NB__: This function is not synthesizable
+--
+-- === __Example with test bench__
+--
+-- @
+-- import Clash.Prelude
+-- import Clash.Explicit.Testbench
+--
+-- import qualified Prelude as P
+--
+-- topEntity
+--   :: 'Signal' 'System' Int
+--   -> 'Signal' 'System' Int
+-- topEntity = id
+--
+-- testBench
+--   :: 'Signal' 'System' Bool
+-- testBench = done
+--  where
+--   testInput = 'Clash.Explicit.Testbench.stimuliGenerator' clk rst $('Clash.Sized.Vector.listToVecTH' [1 :: Int .. 10])
+--   expectedOutput =
+--     'Clash.Explicit.Testbench.outputVerifier'' clk rst $('Clash.Sized.Vector.listToVecTH' $ [1 :: Int .. 9] P.++ [42])
```

Comment:
If the qualified import of `Prelude` is just for `P.++` I'd be inclined to replace it with `<>` which is already exported from `Clash.Prelude`

Reply:
The code is intended for human consumption, it's to show people what the output of `runUntil` looks like given a test bench that they should be able to quickly fathom fully. On the one hand it might be something people need to look up before they understand, on the other hand it's a neat "trick" to teach people in the process, to use the `Semigroup` instance of `List` to avoid needing imports. So I agree it's a good change.